cpmonitor/views.py

```diff
@@ -15,6 +15,37 @@
 from .models import City, ExecutionStatus, Task
 
 
+def _get_children(city, node=None):
+    statuses = {s.value: s for s in ExecutionStatus}
+
+    if not node:
+        children = Task.get_root_nodes()
+    else:
+        children = node.get_children()
+
+    groups = children.filter(city=city, numchild__gt=0)
+    tasks = children.filter(city=city, numchild=0)
+    for group in groups:
+        subtasks = group.get_descendants().filter(numchild=0)
+        total = len(subtasks)
+        group.statuses = {}
+        for subtask in subtasks:
+            status = subtask.execution_status
+            group.statuses[status] = group.statuses.get(status, 0) + round(100 / total)
```

Comment:
This counts how often a certain status appears, and at the same time already converts those numbers to percentages relative to the total number of tasks, but I found it difficult to understand that from just reading these lines.
I would find it much easier to understand if the code would actually mention "count" in some way, and maybe not add up the percentages immediately but count first and then do the ratio calculation. Something like this I find much easier to read:
```suggestion
from collections import Counter
...
        status_counts = Counter(map(lambda subtask: subtask.execution_status, subtasks)
        group_statuses = map(lambda status: counts[status] / counts.total() * 100, counts)
```

Reply:
Now, it took me a while to get your code. ;-) I would prefer list/dict comprehensions over map function, I find them easier to read, i.e. 
`
from collections import Counter

status_counts = Counter([s.execution_status for s in subtasks])
group.task_proportions = {s : c/tasks_count*100 for s, c in status_counts.items()}
`

I still find an explicit loop even easier to read and more efficient though. How about:
`
from collections import defaultdict

single_task_percentage = 1/tasks_count * 100
group.task_proportions = task_proportions = defaultdict(lambda: 0)
for subtask in subtasks:
    task_proportions[subtask.execution_status] += single_task_percentage
`

Pick your choice. I am fine with both.

Reply:
I like @kleins05 first version. Not convinced, the second would be more efficient.

Reply:
The second would certainly be more efficient, as there is only one loop over all tasks. The first version needs to iterate twice, once over the tasks and once over the statuses. However, efficiency is not an issue here for a handful of items, so if you prefer the first, let's take that, it's certainly more pythonic.